src/autoscaler/api/brokerserver/broker_handler.go

```diff
@@ -146,17 +147,25 @@ func (h *BrokerHandler) BindServiceInstance(w http.ResponseWriter, r *http.Reque
 	bindingId := vars["bindingId"]
 	var policyGuid *uuid.UUID
 	body := &models.BindingRequestBody{}
-	err := json.NewDecoder(r.Body).Decode(&body)
+	bodyBytes, err := ioutil.ReadAll(r.Body)
 	if err != nil {
-		h.logger.Error("failed to create binding when trying to read request body", err)
+		h.logger.Error("failed to read bind request body", err, lager.Data{"instanceId": instanceId, "bindingId": bindingId})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Interal-Server-Error",
+			Message: "Failed to read request body"})
+		return
+	}
+	err = json.Unmarshal(bodyBytes, body)
+	if err != nil {
+		h.logger.Error("failed to unmarshal bind body", err, lager.Data{"instanceId": instanceId, "bindingId": bindingId, "body": string(bodyBytes)})
 		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
 			Code:    "Bad Request",
 			Message: "Invalid request body format"})
 		return
 	}
 
-	if body.AppID == "" || instanceId == "" || bindingId == "" || body.ServiceID == "" || body.PlanID == "" {
-		h.logger.Error("failed to create binding when trying to get mandatory data", nil, lager.Data{"appId": body.AppID, "instanceId": instanceId, "bindingId": bindingId, "serviceId": body.ServiceID, "planId": body.PlanID})
+	if body.AppID == "" || instanceId == "" || bindingId == "" {
```

Comment:
according to https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#binding , service ID and planID must be provided here.